src/Shared/Microsoft.AspNetCore.Razor.Test.Common/ConditionalFactAttribute.cs

```diff
@@ -0,0 +1,192 @@
+/// </remarks>
+public sealed class ConditionalTheoryAttribute : TheoryAttribute
+{
+    public ConditionalTheoryAttribute(params string[] conditions)
+    {
+        if (!Conditions.AllTrue(conditions))
+        {
+            base.Skip = Reason ?? Conditions.GetSkipReason(conditions);
+        }
+    }
+
+    /// <summary>
+    ///  This property exists to prevent users of <see cref="ConditionalFactAttribute"/>
+    ///  from accidentally putting documentation in the <see cref="Skip"/> property instead of Reason.
+    ///  Setting <see cref="Skip"/> would cause the test to be unconditionally skip.
+    /// </summary>
+    [Obsolete($"{nameof(ConditionalFactAttribute)} should always use {nameof(Reason)} or {nameof(AlwaysSkip)}", error: true)]
+    public new string Skip
+    {
+        get { return base.Skip; }
+        set { base.Skip = value; }
+    }
+
+    /// <summary>
+    ///  Use to unconditionally skip a test.
+    /// </summary>
+    /// <remarks>
+    ///  This is useful in the rare occasion when a conditional test needs to be skipped unconditionally.
+    ///  Typically, this is for a short term reason, such as working on a bug fix.
+    /// </remarks>
+    public string AlwaysSkip
+    {
+        get { return base.Skip; }
+        set { base.Skip = value; }
+    }
+
+    public string? Reason { get; set; }
+}
+
+public static class Is
+{
+    /// <summary>
+    ///  Only execute if the current operating system platform is Windows.
+    /// </summary>
+    public const string Windows = nameof(Windows);
+
+    /// <summary>
+    ///  Only execute if the current operating system platform is Linux.
+    /// </summary>
+    public const string Linux = nameof(Linux);
+
+    /// <summary>
+    ///  Only execute if the current operating system platform is MacOS.
+    /// </summary>
+    public const string MacOS = nameof(MacOS);
+
+    /// <summary>
+    ///  Only execute if the current operating system platform is FreeBSD.
+    /// </summary>
+    public const string FreeBSD = nameof(FreeBSD);
+
+    public static class Not
+    {
+        /// <summary>
+        ///  Only execute if the current operating system platform is not Windows.
+        /// </summary>
+        public const string Windows = $"!{nameof(Windows)}";
+
+        /// <summary>
+        ///  Only execute if the current operating system platform is Linux.
+        /// </summary>
+        public const string Linux = $"!{nameof(Linux)}";
+
+        /// <summary>
+        ///  Only execute if the current operating system platform is not MacOS.
+        /// </summary>
+        public const string MacOS = $"!{nameof(MacOS)}";
+
+        /// <summary>
+        ///  Only execute if the current operating system platform is not FreeBSD.
+        /// </summary>
+        public const string FreeBSD = $"!{nameof(FreeBSD)}";
+    }
+}
+
+public static class Conditions
+{
+    private static readonly FrozenDictionary<string, Func<bool>> s_conditionMap = CreateConditionMap();
```

Comment:
Why use `Func<bool>` here instead of just `bool`? I can't see an advantage to delaying the evaluation. What am I missing?

Reply:
The only conditions are currently platforms, but I was expecting there to be conditions later that might require more involved checks of the environment. Maybe that's not a concern?

Reply:
That's fine by me